src/test/java/org/opensearch/security/test/helper/rest/RestHelper.java

```diff
@@ -357,6 +362,38 @@ public String toString() {
                     + ", statusReason=" + statusReason + "]";
         }
 
+		/**
+		 * Given a json path with dots delimiated returns the object at the leaf 
+		 */
+		public String findValueInJson(final String jsonDotPath) {
+			// Make sure its json / then parse it
+			if (!isJsonContentType()) {
+				fail("Response was expected to be JSON, body was: \n" + body);
+			}
+			JsonNode currentNode = null;
+			try {
+				currentNode = DefaultObjectMapper.readTree(body);
+			} catch (final Exception e) {
+				throw new RuntimeException(e);
+			}
+
+			// Break the path into parts, and scan into the json object
+			try (final Scanner jsonPathScanner = new Scanner(jsonDotPath).useDelimiter("\\.")) {
+				do {
+					final String pathEntry = jsonPathScanner.next();
```

Comment:
This should be use while loop instead of `do...whilte`. Otherwise, this line could throw exception is `jsonPathScanner` doesn't have next.

Reply:
The only case this would happen is if the string was an empty string, then per the javadoc it would throw `@throws NoSuchElementException if no more tokens are available`.  This seems an appropriate failure that would only happen during test authorship, not in practice after a test has been merged.

Reply:
Great! It's nice to identify the use cases where the exception could happen! Let's say the empty string case, if `NoSuchElementException` gets thrown, the method `findValueInJson` doesn't catch or throw it. And we are not supposed to let this exception happen here at all. It makes more sense to avoid the exception here by validating the condition before the operations unless it has to run the logic in `do` block before evaluating the condition.

Reply:
I think this is a matter of preference, I think the exception is good enough, if this becomes a problem over time we can enhance it.  Is this issue worth blocking the pull request?

Reply:
There's a valid case here as discussed above. When `jsonDotPath` is an empty string, `final String pathEntry = jsonPathScanner.next();` throws `NoSuchElementException` that is not even caught or thrown. I actually don't see a value of using `do...while` here. In fact, it's error-prone. 
This contribution is valuable, it's only a minor fix away from being merged.

Reply:
Did this get marked as resolved in accident? unresolving this conversation

Reply:
This was resolved, see the new lines added above to cover this scenario